src/data_structures.rs

```diff
@@ -104,11 +105,17 @@ pub trait PCProof: Clone + ark_ff::ToBytes + CanonicalSerialize + CanonicalDeser
 
 /// A proof of satisfaction of linear combinations.
 #[derive(Clone, CanonicalSerialize, CanonicalDeserialize)]
-pub struct BatchLCProof<F: Field, P: Polynomial<F>, PC: PolynomialCommitment<F, P>> {
+pub struct BatchLCProof<
+    F: PrimeField,
+    P: Polynomial<F>,
+    PC: PolynomialCommitment<F, P, S>,
+    S: CryptographicSponge,
+> {
     /// Evaluation proof.
     pub proof: PC::BatchProof,
     /// Evaluations required to verify the proof.
     pub evals: Option<Vec<F>>,
+    pub(crate) _sponge: PhantomData<S>,
 }
```

Comment:
To remove the dependence on `Sponge` here (it's not necessary really), maybe we change the definition to the following?
```rust
pub struct BatchLCProof<F: PrimeField, T> {
	pub proof: T,
	pub evals: Option<Vec<F>>,
}
```

I don't recall how this would affect things for users, but maybe we can try it and see.

Reply:
`Sponge` is a type parameter for `PC`. I'm not sure how to remove it...

Reply:
Ah no, I mean removing the `PC` parameter entirely, like suggested in the comment above.

Reply:
got it. Let me have a try

Reply:
looks that there is an issue: doing so, `Marlin::open_combinations` will require explicit type annotation `<_ as PairingEngine>::Fr == <E as PairingEngine>::Fr`, but type annotation is not allowed because of arguments have `impl Trait`. (that is so weird)

Reply:
Can you link me to the code?

Reply:
just pushed that code to PR (won't compile)

Reply:
fixed